### PR TITLE
chore: improve types for HDWallet adapters

### PR DESCRIPTION
**Pull request by cjthompson**
## Description

Use an enum for the list of supported Wallets / KeyManagers. This allows us to avoid casting types in other places.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [X] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [X] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

### CODE REVIEW

File: src/context/WalletProvider/KeepKey/components/Connect.tsx [hunk 1]
```diff
@@ -7,11 +7,10 @@ import {
   ModalHeader
 } from '@chakra-ui/react'
 import { Event } from '@shapeshiftoss/hdwallet-core'
-import { Adapter } from '@shapeshiftoss/hdwallet-keepkey'
 import React, { useState } from 'react'
 import { RouteComponentProps } from 'react-router-dom'
 import { Text } from 'components/Text'
-import { SUPPORTED_WALLETS } from 'context/WalletProvider/config'
+import { KeyMananger, SUPPORTED_WALLETS } from 'context/WalletProvider/config'
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/KeepKey/components/Connect.tsx:13]: mananananger

File: src/context/WalletProvider/SelectModal.tsx [hunk 2]
```diff
@@ -12,26 +17,29 @@ export const SelectModal = ({ connect }: { connect: IWalletContext['connect'] })
...
-                justifyContent='space-between'
-                onClick={() => connect(key)}
-              >
-                <Text fontWeight='semibold'>{option.name}</Text>
-                <Icon height='auto' w='45px' />
-              </Button>
-            )
-          })}
+          {adapters &&
+            // Only iterate over the adapters that successfully initialized
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/SelectModal.tsx:21]: we should probably render supported but unavailable wallets differently with a tooptip or similar giving the user feedback as to why a plugged in keepkey might be unavailable, but we can do that in a separate PR

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
please rename `KeyMananger` -> `KeyManager`

File: src/context/WalletProvider/SelectModal.tsx (hunk 2 above)
Comment (cjthompson) [reply src/context/WalletProvider/SelectModal.tsx:21]: I can revert this change.

**Comment by cjthompson** [comment]
> please rename `KeyMananger` -> `KeyManager`

Ironically I fixed another typo in this PR for `initial`
